lib/matplotlib/axes/_axes.py

```diff
@@ -940,12 +948,28 @@ def axline(self, xy1, xy2, **kwargs):
         axhline : for horizontal lines
         axvline : for vertical lines
         """
+        def _to_points(xy1, xy2, slope):
+            """
+            Check for a valid combination of input parameters and convert
+            to two points, if necessary.
+            """
+            if (xy2 is None and slope is None or
+                    xy2 is not None and slope is not None):
+                raise TypeError(
+                    "Exactly one of 'xy2' and 'slope' must be given")
+            if xy2 is None:
+                x1, y1 = xy1
+                xy2 = x1 + 1, y1 + slope
```

Comment:
this case should definitely not include the "second" point (x2, y2) in the datalim, as that's now an internal helper, not a user-provided point

Reply:
done.